ramalama/gpu_detector.py

```diff
@@ -0,0 +1,168 @@
+
+    def get_intel_gpu(self):
+        """Detects Intel GPUs using sysfs and lspci (Linux) or system_profiler (macOS)."""
+        if platform.system() == "Linux":
+            self._read_gpu_memory('/sys/class/drm/card*/device/mem_info_vram_total', "Intel GPU", "ONEAPI_DEVICE_SELECTOR")
+
+            # Alternative detection method using lspci
+            try:
+                result = subprocess.run(["lspci", "-nn"], capture_output=True, text=True, check=True)
+                output = result.stdout.strip()
+                if "Intel Corporation" in output:
+                    if self.best_gpu is None:  # If no VRAM info was found
+                        self.best_gpu = "Intel GPU (unknown memory)"
+                        self.best_env = "ONEAPI_DEVICE_SELECTOR"
+            except FileNotFoundError:
+                raise RuntimeError("`lspci` not found. Unable to detect Intel GPU.")
+            except subprocess.CalledProcessError as e:
+                error_msg = e.stderr.strip() if e.stderr else "Unknown error (check if `pciutils` is installed)."
+                raise RuntimeError(f"Unable to detect Intel GPU(s). Error: {error_msg}")
+        elif platform.system() == "Darwin":  # macOS
+            self.get_macos_gpu()  # macOS detection covers Intel GPUs
+
+    def get_macos_gpu(self):
+        """Detects GPUs on macOS using system_profiler."""
+        try:
+            result = subprocess.run(["system_profiler", "SPDisplaysDataType"], capture_output=True, text=True, check=True)
+            output = result.stdout.strip()
+
+            gpu_name = None
+            vram_mib = 0  # Default to unknown
+
+            for line in output.split('\n'):
+                line = line.strip()
+                if line.startswith("Chipset Model:"):
+                    gpu_name = line.split(":")[1].strip()
+                elif line.startswith("VRAM") or line.startswith("VRAM (Dynamic)"):
+                    try:
+                        vram_mib = int(line.split(":")[1].strip().replace("MB", ""))
+                        self._update_best_gpu(vram_mib, gpu_name, "METAL_DEVICE_INDEX")
+                    except ValueError:
+                        continue  # Skip invalid VRAM values
+
+            # Handle Apple Silicon (M1, M2, etc.)
+            if gpu_name and "Apple" in gpu_name:
+                self._update_best_gpu(8192, gpu_name, "METAL_DEVICE_INDEX")  # Assume 8GB VRAM for M1/M2 Max
+
+        except FileNotFoundError:
+            raise RuntimeError("`system_profiler` not found. Unable to detect macOS GPU.")
+        except subprocess.CalledProcessError as e:
+            error_msg = e.stderr.strip() if e.stderr else "Unknown error."
+            raise RuntimeError(f"Unable to detect macOS GPU(s). Error: {error_msg}")
+
+    def detect_best_gpu(self, gpu_template):
+        """Compares Nvidia, AMD, and Intel GPUs and appends the best GPU with the highest VRAM to gpu_template.
```

Comment:
What about MAC gpu?

Reply:
It works, let me update. 

Linux:
```
ramalama --show-gpus-available 
Warning: Unable to detect NVIDIA GPU(s). Error: Unknown error (check if NVIDIA drivers are loaded).
GPU(s) detected:
  - Index: Intel GPU (unknown memory), VRAM: 0 MiB, Env: ONEAPI_DEVICE_SELECTOR
```

macOS:
```
ramalama --show-gpus-available
GPU(s) detected:
  - Index: Apple M1 Max, VRAM: 8192 MiB, Env: METAL_DEVICE_INDEX
```